File: azure-functions/sample_codes/getting-started/http-trigger.ts
// Azure Functions HTTP Trigger - TypeScript Node.js v4 Programming Model
//
// Prerequisites:
//   npm install @azure/functions
//
// Run locally: npm start (or func start)
// Test: GET http://localhost:7071/api/httpTrigger?name=World
//       POST http://localhost:7071/api/httpTrigger (body: "World")

import {
  app,
  HttpRequest,
  HttpResponseInit,
  InvocationContext,
} from "@azure/functions";

export async function httpTrigger(
  request: HttpRequest,
  context: InvocationContext
): Promise<HttpResponseInit> {
  context.log(`Http function processed request for url "${request.url}"`);

  const name =
    request.query.get("name") || (await request.text()) || "world";

  return { body: `Hello, ${name}!` };
}

app.http("httpTrigger", {
  methods: ["GET", "POST"],
  authLevel: "anonymous",
  handler: httpTrigger,
});

File: azure-functions/sample_codes/getting-started/http-trigger.py
# Azure Functions HTTP Trigger - Python v2 Programming Model
#
# Prerequisites:
#   pip install azure-functions
#
# Run locally: func start
# Test: GET http://localhost:7071/api/HttpExample?name=World
#       POST http://localhost:7071/api/HttpExample (body: {"name": "World"})

import azure.functions as func
import logging
import json

app = func.FunctionApp(http_auth_level=func.AuthLevel.ANONYMOUS)


@app.route(route="HttpExample")
def http_example(req: func.HttpRequest) -> func.HttpResponse:
    logging.info("Python HTTP trigger function processed a request.")

    # Read name from query string
    name = req.params.get("name")

    # Or read from request body
    if not name:
        try:
            req_body = req.get_json()
            name = req_body.get("name")
        except ValueError:
            pass

    if name:
        return func.HttpResponse(
            f"Hello, {name}!",
            status_code=200
        )
    else:
        return func.HttpResponse(
            "Pass a name in the query string or request body.",
            status_code=200
        )
